### PR TITLE
add DDAgentExtraEnvVars

### PR DESCRIPTION
**Pull request by kangyili**
What does this PR do?
---------------------

- Add a new config entry point `DDAgentExtraEnvVars` to allow users to pass into environment variables

Which scenarios this will impact?
-------------------

Motivation
----------

Additional Notes
----------------

### CODE REVIEW

File: common/config/environment.go [hunk 1]
```diff
@@ -45,6 +46,7 @@ const (
 	DDAgentAPPKeyParamName               = "appKey"
 	DDAgentFakeintake                    = "fakeintake"
 	DDAgentSite                          = "site"
+	DDAgentExtraEnvVars                  = "extraEnvVars" // extraEnvVars is expected in the format: <key1>=<value1>,<key2>=<value2>,...
```
Comment (kangyili) [inline common/config/environment.go:49]: example:
https://github.com/DataDog/datadog-agent/blob/b8e0f05d5404d66c526d3dc43bd4bb8be2c1ea46/test/new-e2e/tests/orchestrator/ecs_test.go#L65

File: resources/aws/ecs/fargateService.go [hunk 2]
```diff
@@ -46,7 +47,7 @@ func FargateTaskDefinitionWithAgent(
 	fakeintake *fakeintake.Fakeintake,
 	opts ...pulumi.ResourceOption,
 ) (*ecs.FargateTaskDefinition, error) {
-	containers["datadog-agent"] = *agent.ECSFargateLinuxContainerDefinition(*e.CommonEnvironment, "public.ecr.aws/datadog/agent:latest", apiKeySSMParamName, fakeintake, GetFirelensLogConfiguration(pulumi.String("datadog-agent"), pulumi.String("datadog-agent"), apiKeySSMParamName))
+	containers["datadog-agent"] = *agent.ECSFargateLinuxContainerDefinition(*e.CommonEnvironment, "", apiKeySSMParamName, fakeintake, GetFirelensLogConfiguration(pulumi.String("datadog-agent"), pulumi.String("datadog-agent"), apiKeySSMParamName))
```
Comment (kangyili) [inline resources/aws/ecs/fargateService.go:50]: remove image here so we can provide it by `e.CommonEnvironment`
https://github.com/DataDog/test-infra-definitions/blob/77b752d40abb3c9cc919d0061ed9f2f1105a8101/components/datadog/agent/ecsFargate.go#L12-L16

**Review by FlorentClarret — COMMENTED** [review]
Thanks for this PR! I left one suggestion to make the code easier to re-use. LEt me know what you think

File: common/config/environment.go [hunk 3]
```diff
@@ -275,6 +278,10 @@ func (e *CommonEnvironment) Site() string {
 	return e.AgentConfig.Get(DDAgentSite)
 }
 
+func (e *CommonEnvironment) AgentExtraEnvVars() string {
+	return e.AgentConfig.Get(DDAgentExtraEnvVars)
+}
```
Comment (FlorentClarret) [inline common/config/environment.go:283]: **💬 suggestion:** ‏

Could you parse the string in this function and return a `map[string]string` so it's easier to re-use it somewhere else in the code?

Comment (kangyili) [reply common/config/environment.go]: [dc91e32](https://github.com/DataDog/test-infra-definitions/pull/801/commits/dc91e32be9526f573dbc8434b0956bb665377486)

**Review by FlorentClarret — APPROVED** [review]

**Pull request merged**